src/tests/rules/boolean-keywords.coffee

```diff
@@ -0,0 +1,138 @@
+###*
+# @fileoverview This rule should require or disallow usage of specific boolean keywords.
+# @author Julian Rosse
+###
+'use strict'
+
+#------------------------------------------------------------------------------
+# Requirements
+#------------------------------------------------------------------------------
+
+rule = require '../../rules/boolean-keywords'
+{RuleTester} = require 'eslint'
+path = require 'path'
+
+error = (unexpected, replacement) ->
+  if replacement?
+    messageId: 'unexpected-fixable'
+    data: {
+      unexpected
+      replacement
+    }
+  else
+    messageId: 'unexpected'
+    data: {
+      unexpected
+    }
+
+#------------------------------------------------------------------------------
+# Tests
+#------------------------------------------------------------------------------
+
+ruleTester = new RuleTester parser: path.join __dirname, '../../..'
+
+ruleTester.run 'boolean-keywords', rule,
+  valid: [
+    code: 'yes or no'
+    options: [allow: ['yes', 'no']]
+  ,
+    code: 'on or off'
+    options: [allow: ['on', 'off']]
+  ,
+    code: 'true or false'
+    options: [allow: ['true', 'false']]
+  ,
+    code: 'yes or no'
+    options: [disallow: ['true', 'false']]
+  ,
+    code: 'yes or no'
+    options: [disallow: ['on', 'off']]
+  ,
+    code: 'yes or no'
+    options: [disallow: ['on', 'off', 'true']]
+  ,
+    code: 'on or off'
+    options: [disallow: ['yes', 'no']]
+  ,
+    code: 'on or off'
+    options: [disallow: ['true', 'false']]
+  ,
+    code: 'true'
+    options: [disallow: ['yes', 'no']]
+  ,
+    code: 'false'
+    options: [disallow: ['on', 'off']]
+  ]
+
+  invalid: [
+    code: 'a is true'
+    output: 'a is yes'
```

Comment:
For fixable rules, you can also specify the expected `output` for code snippets that should trigger the rule to specify a fix